File: app/src/main/java/it/liceoarzignano/bold/firebase/BoldInstanceIdService.java
package it.liceoarzignano.bold.firebase;

import android.content.Context;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;
import com.google.firebase.messaging.FirebaseMessaging;

import it.liceoarzignano.bold.Utils;


public class BoldInstanceIdService extends FirebaseInstanceIdService {
    private static final String ADDR1_TOPIC = "Scientifico";
    private static final String ADDR2_TOPIC = "ScApplicate";
    private static final String ADDR3_TOPIC = "Linguistico";
    private static final String ADDR4_TOPIC = "ScUmane";
    private static final String ADDR5_TOPIC = "Economico";
    private static final String ADDR6_TOPIC = "Docente";

    @Override
    public void onTokenRefresh() {
        FirebaseInstanceId.getInstance().getToken();
        Context context = getApplicationContext();
        String topic;

        if (Utils.isTeacher(context)) {
            topic = ADDR6_TOPIC;
        } else {
            switch (Utils.getAddress(context)) {
                case "1":
                    topic = ADDR1_TOPIC;
                    break;
                case "2":
                    topic = ADDR2_TOPIC;
                    break;
                case "3":
                    topic = ADDR3_TOPIC;
                    break;
                case "4":
                    topic = ADDR4_TOPIC;
                    break;
                case "5":
                    topic = ADDR5_TOPIC;
                    break;
                default:
                    topic = ADDR6_TOPIC;
                    break;
            }
        }
        FirebaseMessaging.getInstance().subscribeToTopic(topic);
    }
}

File: app/src/main/kotlin/it/liceoarzignano/bold/firebase/BoldInstanceIdService.kt
package it.liceoarzignano.bold.firebase

import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.iid.FirebaseInstanceIdService
import com.google.firebase.messaging.FirebaseMessaging
import it.liceoarzignano.bold.settings.AppPrefs


class BoldInstanceIdService : FirebaseInstanceIdService() {

    override fun onTokenRefresh() {
        FirebaseInstanceId.getInstance().token
        val prefs = AppPrefs(baseContext)
        val topic: String

        topic = if (prefs.get(AppPrefs.KEY_IS_TEACHER)) {
            ADDR6_TOPIC
        } else {
            when (prefs.get(AppPrefs.KEY_ADDRESS, "0")) {
                "1" -> ADDR1_TOPIC
                "2" -> ADDR2_TOPIC
                "3" -> ADDR3_TOPIC
                "4" -> ADDR4_TOPIC
                "5" -> ADDR5_TOPIC
                else -> ADDR6_TOPIC
            }
        }
        FirebaseMessaging.getInstance().subscribeToTopic(topic)
    }

    companion object {
        private const val ADDR1_TOPIC = "Scientifico"
        private const val ADDR2_TOPIC = "ScApplicate"
        private const val ADDR3_TOPIC = "Linguistico"
        private const val ADDR4_TOPIC = "ScUmane"
        private const val ADDR5_TOPIC = "Economico"
        private const val ADDR6_TOPIC = "Docente"
    }
}
